core/src/test/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpgradeManagementServiceTest.java

```diff
@@ -0,0 +1,172 @@
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.TestUtil;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.cloud.model.ClouderaManagerRepo;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterApi;
+import com.sequenceiq.cloudbreak.cluster.service.ClusterComponentConfigProvider;
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade.ClusterUpgradeService;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorException;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.cloudbreak.service.upgrade.sync.component.CmServerQueryService;
+
+@ExtendWith(MockitoExtension.class)
+public class ClusterManagerUpgradeManagementServiceTest {
+
+    private static final Long STACK_ID = 1L;
+
+    private static final String OLD_CM_VERSION = "7.2.2-12345";
+
+    private static final String CM_VERSION = "7.2.6-12345";
+
+    private static final String CM_VERSION_WITH_P = "7.2.6-12345p";
+
+    private static final String STACK_VERSION = "7.2.7";
+
+    @Mock
+    private StackDtoService stackDtoService;
+
+    @Mock
+    private ClusterComponentConfigProvider clusterComponentConfigProvider;
+
+    @Mock
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Mock
+    private ClusterApi clusterApi;
+
+    @Mock
+    private CmServerQueryService cmServerQueryService;
+
+    @Mock
+    private ClusterUpgradeService clusterUpgradeService;
+
+    @Mock
+    private ClusterManagerUpgradeService clusterManagerUpgradeService;
+
+    @InjectMocks
+    private ClusterManagerUpgradeManagementService underTest;
+
+    @Spy
+    private StackDto stackDto;
+
+    private Stack stack;
+
+    private Cluster cluster;
+
+    private static Stream<Arguments> cmVersions() {
+        return Stream.of(
+                Arguments.of(CM_VERSION, CM_VERSION),
+                Arguments.of(CM_VERSION_WITH_P, CM_VERSION),
+                Arguments.of(CM_VERSION, CM_VERSION_WITH_P),
+                Arguments.of(CM_VERSION_WITH_P, CM_VERSION_WITH_P)
+        );
+    }
+
+    @BeforeEach
+    public void setUp() {
+        stack = TestUtil.stack(Status.AVAILABLE, TestUtil.awsCredential());
+        cluster = TestUtil.cluster();
+        when(stackDtoService.getById(STACK_ID)).thenReturn(stackDto);
+        //when(clusterApiConnectors.getConnector(stackDto)).thenReturn(clusterApi);
+        //when(stackDto.getStack()).thenReturn(stack);
+//        when(stackDto.getStackVersion()).thenReturn(STACK_VERSION);
```

Comment:
if these are not needed, could you remove them?

Reply:
Sure, thanks